### ISSUE TITLE
The installer says  "macOS Sierra is already installed on this Mac."

### ISSUE THREAD

**Issue by Rocket1184** [syscl/XPS9350-macOS#95]
It's my 1st time to try hackintosh, maybe something went wrong....

## Laptop hardware:
```
CPU: Core i7 6560U
Ram: 16GB
SSD: 512GB NVMe
Screen: 3200x1800
```

I've already changed the BIOS settings cause I have only Linux installed on my laptop.
and I used a MacBook from my roommate to create the install media

## My installation steps:
1. parted a 16GB USB disk to 2 partitions: 200MiB FAT for EFI partition, and other for HFS+
2. install clover v2.4k_r4049 with UEFI only and Drivers64UEFI
3. delte the CLOVER folder and copy the CLOVER folder in the repo there
4. create the install media with:

```
sudo /Applications/Install\ macOS\ Sierra.app/Contents/Resources/createinstallmedia --volume /Volumes/install_osx --applicationpath /Applications/Install\ macOS\ Sierra.app --nointeraction
```
the installer version was `12.0.49(1249)`

5. reboot, add the CLOVER to the boot options, boot into USB install disk
6. press F4, and the mouse lagged for servel seconds....
7. start the install with arg `-v`, and the 1st error occurs: 

```
OsxAptioFix2Drv: Starting overrides for \.IABootFiles\boot.efi
Using reloc block: no, hibernate wake: no

***********************************************************
This version of Mac OS X is not supported on this platform!
***********************************************************
Reason: Mac-66e35819ee2d0d05
```

that's so confused...after some google, i got one solution:
add argument `-no_compat_check`

then I can enter the installation GUI.
8. format the disk using the disk utils provide by installation GUI, when I'm about to select the disk, 2nd problem occurs, the installer says 

```
macOS Sierra is already installed on this Mac.
```

then i exit the installation process, format the disk again with HFS+J with or without encrypt, but it still complaints about that

![img_20170414_183238](https://cloud.githubusercontent.com/assets/13914967/25045844/d4756c38-20f3-11e7-9e96-a20787b1093b.jpg)

do you have any ideas about that?

**Comment by flipphos** [comment]
I have the exact laptop as you do, and the latest macOS works fine on mine. So don’t worry, you will succeed.

Could you try to erase you SSD and partition then format with HFS+J before you enter the installation. I did this on the whole SSD when I installed mine.

Good luck!

**Comment by Rocket1184** [comment]
@flipphos 
Did you get the 'unsupported platform' warning when installing?
I'm not sure if it's right to add that boot argument....

thanks

**Comment by flipphos** [comment]
@Rocka
No, I didn't get the warning and neither used that boot argument.
Are you sure you copied the 'Clover' folder from the repo to your USB stick and the file 'Fakesmc' is in place?
Did you swap your wifi card? I replaced the stock card with DW1560.

**Comment by Rocket1184** [comment]
@flipphos 
I'm sure i copied the whole CLOVER folder to the EFI partition of my USB disk. but what dose 'Fakesmc' mean? where is it....
I didn't swap the wifi card, i use the stock one

thanks

**Comment by flipphos** [comment]
I meant fakesmc.kext which you may find under /clover/kexts/others
You can’t install macOS without this kext file.

**Comment by flipphos** [comment]
@Rocka
Like this:

**Comment by Rocket1184** [comment]
@flipphos
yes, i can find `FakeSMC.kext` under `EFI/CLOVER/kexts/10.12` and `EFI/CLOVER/kexts/10.12`

maybe i would backup my data and erase the disk to try again...but i really could take a long time 

thanks

**Comment by flipphos** [comment]
@Rocka
I guess so, it is always easier to install a Linux later than the macOS.

**Comment by b50** [comment]
What does it say if you run "diskutil list" from the terminal? What model SDD do you have?

**Comment by Rocket1184** [comment]
@b50 
would you please tell me how to open the terminal window during the install process....I'm new to this
and my SSD model is  Toshiba THNSN5512GPU7NV

thanks

**Comment by b50** [comment]
![launch-terminal-from-recovery-hd](https://cloud.githubusercontent.com/assets/583418/25072011/e1939106-2292-11e7-8557-eea8a55fe736.jpg)

**Comment by Rocket1184** [comment]
@b50 
my `diskutil list` output is:

```
/dev/disk0 (internal):
   #:                        TYPE NAME                   SIZE          IDENTIFIER
   0:       GUID_partition_scheme                        512.1 GB      disk0
   1:                         EFI NO NAME                536.9 MB      disk0s1
   2:                  Linux Swap                        1.6 GB        disk0s2
   3:            Linux Filesystem                        4.3 GB        disk0s3
   4:            Linux Filesystem                        214.7 GB      disk0s4

/dev/disk1 (external, physical):
   #:                        TYPE NAME                   SIZE          IDENTIFIER
   0:       GUID_partition_scheme                       *15.6 GB       disk1
   1:                         EFI EFI                    536.9 MB      disk1s1
   2:                   Apple_HFS Install macOS Sierra   15.3 GB       disk1s2

(disk images below....)
```
this is my linux partitons:
efi(fat32), swap, boot(ext4), root(f2fs)

I deleted the HFS+ partition after that failure
hopes that info may help...

thanks

**Comment by b50** [comment]
Ok that all looks fine. I have a few things you could try:
1. Add another flash drive or external hard drive and see if it would allow you to install on one of those. If it does, it's probably a problem with your SSD drivers.
3. Try add [SATA-100-series-unsupported.kext](https://www.dropbox.com/sh/3lctmui3zkflpdl/AAC1TZw-KM-7JNK8QQji5hUka?dl=1) to your to clover kexts. It doesn't look like your SSD would benefit, but definitely worth a try.
4. Try replacing your USB efi partition with [this](https://www.dropbox.com/s/vefjl29j6jpr07q/USB%20EFI%20Partition.zip?dl=1). That's my EFI partition that I used to install macOS on my XPS.
5. Reinstall macOS Sierra onto your USB or even a different USB. 
6. Wipe your SSD completely and see if you could install then. Remember to backup your linux partitions!

**Comment by gjabell** [comment]
I had the same issue as you, and using [this](https://github.com/RehabMan/OS-X-Clover-Laptop-Config/blob/master/config_HD520_530_540.plist) clover config instead of the one from syscl let me boot without the -no_compat_check flag, which I think was causing the problem.

**Comment by Rocket1184** [comment]
@gjabell 
I can boot with the config.plist you mentioned, but it cannot recognize my NVMe SSD...
maybe NVMe driver not included in that config?

**Comment by Rocket1184** [comment]
@C9HDN 
I have tried the EFI partition, it can boot with the `-no_compat_check` flag, but still refuse to install on my SSD... and it also refuse to install on a external HDD drive. 

[this config.plist](https://github.com/RehabMan/OS-X-Clover-Laptop-Config/blob/master/config_HD520_530_540.plist) mentioned by @gjabell can boot perfectly but just cannot recognize the SSD. maybe I would try to merge them?

by the way, which model of XPS are you using ?

thank you all ! !

**Comment by gjabell** [comment]
I am using a 13" 9350 w/ an NVME ssd. How are you making your boot drive? I was having some issues installing for other reasons, and using Unibeast to create the drive worked better for me than default clover.

**Comment by frankrobert** [comment]
@Rocket1184 @gjabell Has this been solved? I have an NVMe Samsung 256GB Drive that has similar issues to @Rocket1184 . Depending on the config I use to install it will get recognised but have the `macOS Sierra is already installed on this Mac` issue or it won't be able to see the drive at all.

I've tried doing a vanilla install via command line -> replacing clover folder. No luck. Gives me the already installed issue and I can only access this with the `-no_compat_check` flag.
Replacing the plist file that @gjabell linked allows me to get to the installer without the compat_check flag but it can't see my drive under these circumstances.

Using Unibeast instead of vanilla install gives me near-identical results.

Thoughts? I'm extremely new to the hackintosh community so if there's anything else missing please let me know. I'd like to contribute, even if it's something like documentation to make this easier to other people in the future.

**Comment by Rocket1184** [comment]
@frankrobert it haven't been sloved yet. I'm preparing for some exam those days and did not have much time on this...
hope you could find an elegant solution !

thanks

**Comment by gjabell** [comment]
@frankrobert What wifi card are you using? Some cards don't work out of the box, and require additional kexts to function, so that may have been your issue with the first edit.

**Comment by syscl** [comment]
@frankrobert you said my settings was "completely outdated", but excuse me, I just used my old settings and updated to latest macOS from App Store without any problems. So I am quiet confused your meaning.

A good hack should be the following:
- Necessary patches for system
- Patches can be applied to future settings
- You tell me

Recently I just finished my exams and summer courses will come, I don't have so much time to update my github. But in the meantime, less update always means this project is almost finished(I don't experience any issues for about 2 mouths).

So if you feel this project is completely outdated, please use pull your merge. But IMO, this project still works as expect.

syscl

**Comment by flipphos** [comment]
Yes, I am happy to second syscl. I just upgaded my macOS to the latest 10.12.5 today morning without doing anything extra. Downloaded from the appstore and it upgraded by itself, as simple as that.

This project rocks!

Best Regards
